core/src/main/java/io/questdb/griffin/engine/table/AsyncGroupByNotKeyedRecordCursorFactory.java

```diff
@@ -0,0 +1,194 @@
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.GroupByFunction;
+import io.questdb.griffin.engine.groupby.GroupByFunctionsUpdater;
+import io.questdb.griffin.engine.groupby.SimpleMapValue;
+import io.questdb.mp.SCSequence;
+import io.questdb.std.BytecodeAssembler;
+import io.questdb.std.Misc;
+import io.questdb.std.ObjList;
+import io.questdb.std.Transient;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static io.questdb.cairo.sql.DataFrameCursorFactory.ORDER_ASC;
+import static io.questdb.cairo.sql.DataFrameCursorFactory.ORDER_DESC;
+
+public class AsyncGroupByNotKeyedRecordCursorFactory extends AbstractRecordCursorFactory {
+
+    private static final PageFrameReducer REDUCER = AsyncGroupByNotKeyedRecordCursorFactory::aggregate;
+    private final AsyncGroupByNotKeyedAtom atom;
+    private final RecordCursorFactory base;
+    private final SCSequence collectSubSeq = new SCSequence();
+    private final AsyncGroupByNotKeyedRecordCursor cursor;
+    private final PageFrameSequence<AsyncGroupByNotKeyedAtom> frameSequence;
+    private final ObjList<GroupByFunction> groupByFunctions;
+    private final ObjList<Function> recordFunctions;
+    private final int workerCount;
+
+    public AsyncGroupByNotKeyedRecordCursorFactory(
+            @Transient @NotNull BytecodeAssembler asm,
+            @NotNull CairoConfiguration configuration,
+            @NotNull MessageBus messageBus,
+            @NotNull RecordCursorFactory base,
+            @NotNull RecordMetadata groupByMetadata,
+            ObjList<GroupByFunction> groupByFunctions,
+            ObjList<Function> recordFunctions,
+            int valueCount,
+            @Nullable Function filter,
+            @NotNull PageFrameReduceTaskFactory reduceTaskFactory,
+            @Nullable ObjList<Function> perWorkerFilters,
+            int workerCount
+    ) {
+        super(groupByMetadata);
+        try {
+            assert !(base instanceof AsyncGroupByNotKeyedRecordCursorFactory);
+            this.base = base;
+            this.groupByFunctions = groupByFunctions;
+            this.recordFunctions = recordFunctions;
+            this.atom = new AsyncGroupByNotKeyedAtom(
+                    asm,
+                    configuration,
+                    groupByFunctions,
+                    valueCount,
+                    filter,
+                    perWorkerFilters,
+                    workerCount
+            );
+            this.frameSequence = new PageFrameSequence<>(configuration, messageBus, REDUCER, reduceTaskFactory, PageFrameReduceTask.TYPE_GROUP_BY_NOT_KEYED);
+            this.cursor = new AsyncGroupByNotKeyedRecordCursor(groupByFunctions, recordFunctions, valueCount);
+            this.workerCount = workerCount;
+        } catch (Throwable e) {
+            Misc.freeObjList(recordFunctions);
```

Comment:
`groupByFunctions` are included in `recordFunctions`, so it would be a double close. Changed to `close()` call in cf57b6d23bbea8fb211c2d61171fddb11034a7ad